gp/primitives.py

```diff
@@ -44,7 +50,7 @@ def _determine_max_arity(self):
 
         self._max_arity = arity
 
-    def sample(self, rng):
+    def sample(self, rng: np.random.RandomState) -> int:
```

Comment:
```suggestion
    def sample_index(self, rng: np.random.RandomState) -> int:
```

or something similar, this name is confusing! although maybe not part of this PR, strictly speaking ;)

Reply:
Yes, out of scope. Create an issue: #106